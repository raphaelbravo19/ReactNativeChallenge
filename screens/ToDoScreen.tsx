import React from "react";
import { StyleSheet, Text, View, Image } from "react-native";
import { Button } from "../components/ui";

const HomeIllustration = require("../assets/home-illustration.png");

export default function ToDoScreen() {
  return (
    <View style={styles.container}>
      <Image style={styles.illustration} source={HomeIllustration} />
      <Text style={styles.title}>Howdy partner!</Text>
      <View style={styles.subtitleContainer}>
        <Text style={styles.subtitle}>
          {
            "This is your assignment.\nFollow the instructions on the Readme file."
          }
        </Text>
        <Text style={styles.subtitle}>
          Don’t worry, it’s easy! You should have the app looking smooth in no
          time.
        </Text>
      </View>
      <Button style={styles.button}>Get Started</Button>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: "#fff",
    alignItems: "center",
    justifyContent: "center",
  },
  illustration: {
    width: 206,
    height: 194,
    marginBottom: 12,
  },
  title: {
    fontSize: 24,
    fontWeight: "bold",
    marginVertical: 12,
  },
  subtitle: {
    fontSize: 14,
    fontWeight: "400",
    textAlign: "center",
    marginTop: 12,
  },
  subtitleContainer: {
    alignItems: "center",
    paddingHorizontal: 30,
  },
  button: {
    alignSelf: "stretch",
    marginHorizontal: 35,
    marginTop: 44,
    marginBottom: 150,
  },
});
