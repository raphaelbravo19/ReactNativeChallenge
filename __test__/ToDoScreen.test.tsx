import React from "react";
import renderer from "react-test-renderer";
import ToDoScreen from "../screens/ToDoScreen";

test("todo renders", () => {
  const tree = renderer.create(<ToDoScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});
