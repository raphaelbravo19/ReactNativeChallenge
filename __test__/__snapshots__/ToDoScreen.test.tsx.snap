// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`todo renders 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../assets/home-illustration.png",
      }
    }
    style={
      Object {
        "height": 194,
        "marginBottom": 12,
        "width": 206,
      }
    }
  />
  <Text
    style={
      Object {
        "fontSize": 24,
        "fontWeight": "bold",
        "marginVertical": 12,
      }
    }
  >
    Howdy partner!
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "paddingHorizontal": 30,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 14,
          "fontWeight": "400",
          "marginTop": 12,
          "textAlign": "center",
        }
      }
    >
      This is your assignment.
Follow the instructions on the Readme file.
    </Text>
    <Text
      style={
        Object {
          "fontSize": 14,
          "fontWeight": "400",
          "marginTop": 12,
          "textAlign": "center",
        }
      }
    >
      Don’t worry, it’s easy! You should have the app looking smooth in no time.
    </Text>
  </View>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "stretch",
        "backgroundColor": "#1FC4DB",
        "borderRadius": 6,
        "marginBottom": 150,
        "marginHorizontal": 35,
        "marginTop": 44,
        "opacity": 1,
        "paddingVertical": 12,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#FFFFFF",
          "fontSize": 16,
          "fontWeight": "500",
        }
      }
    >
      Get Started
    </Text>
  </View>
</View>
`;
